ui/pages/send/send-content/send-hex-data-row/send-hex-data-row.component.test.js

```diff
@@ -0,0 +1,54 @@
+import sinon from 'sinon';
+import React from 'react';
+import { shallow } from 'enzyme';
+import SendRowWrapper from '../send-row-wrapper/send-row-wrapper.component';
+import SendHexDataRow from './send-hex-data-row.component';
+
+require('./send-hex-data-row.container.js');
+
+const updateSendHexData = sinon.spy();
+
+function shallowRenderSendHexDataRow() {
+  const wrapper = shallow(
+    <SendHexDataRow updateSendHexData={updateSendHexData} />,
+    {
+      context: { t: (str) => `${str}_t` },
+    },
+  );
+  const instance = wrapper.instance();
+
+  return {
+    instance,
+    wrapper,
+    propsMethodSpies: {
+      updateSendHexData,
+    },
+  };
+}
+
+describe('send-hex-data-row container', () => {
+  describe('render', () => {
+    it('should render a SendHexData component', () => {
+      const { wrapper } = shallowRenderSendHexDataRow();
+
+      expect(wrapper.find(SendRowWrapper)).toHaveLength(1);
+    });
+
+    it('should pass the correct props to SendRowWrapper', () => {
+      const { wrapper } = shallowRenderSendHexDataRow();
+      const { errorType, label, showError } = wrapper
+        .find(SendRowWrapper)
+        .props();
+
+      expect(errorType).toStrictEqual('amount');
+      expect(label).toStrictEqual('hexData_t:');
+      expect(showError).toBeUndefined();
+    });
+
+    it('should render a textarea', () => {
+      const { wrapper } = shallowRenderSendHexDataRow();
+
+      expect(wrapper.find('textarea').exists()).toStrictEqual(true);
+    });
+  });
```

Comment:
Good one adding the test cases, but you're missing testing for events when an in/valid text (non-hex) is inputed.
i. should properly handle invalid hex input
ii. should properly handle valid hex input

Reply:
so I converted the component to functional and used react-testing-library instead. fixed [here](https://github.com/MetaMask/metamask-extension/pull/14724/commits/6d6883b2e37463e18a7a5890201b47e09da9ae3f)